connectionmanager/connectionmanager.go

```diff
@@ -0,0 +1,118 @@
+package connectionmanager
+
+import (
+	"log"
+	"sync"
+)
+
+// Initializer is the interface that initializes a connection of some kind.
+type Initializer interface {
+	Initialize()
+}
+
+// Closer is the interface that gracefully closes a connection of some kind.
+type Closer interface {
+	Close()
+}
+
+// InitializerCloser is the interface the both initializes and gracefully closes a
+// connection of some kind.
+type InitializerCloser interface {
+	Initializer
+	Closer
+}
+
+// CloseFunc is a function that gracefully shuts down a connection as a side effect.
+type CloseFunc func()
+
+// ConnectionManager allows multiple items needing initialization or shutdown to be
+// managed as a group.
+//
+// Initialization and Close of connections are managed independently of each other. Once
+// created the connection manager can accept any number of items supporting initialization,
+// close, or both. The ManageInitialization, ManageClose, and ManageConnection methods can
+// be called as many times as needed in any order to add managed items. They are variadic
+// functions, so multiple items can be added in a single call.
+//
+// Initialization items will immediately start initialization in a separate go routine
+// once the item is added to the ConnectionManager. An internally managed sync.WaitGroup
+// is made available. Calling WG.Wait() on the ConnectionManager will block the current
+// go routine until all initialization functions are complete.
+//
+// Closers are queued up internally running only when the ConnectionManager's Close method
+// is called. The ConnectionManager runs each Close method in a separate go routine and blocks
+// blocks until all are complete.
+type ConnectionManager struct {
```

Comment:
I _believe_ a more idiomatic naming scheme would be to remove `Connection` prefix from the names within this package to avoid the stutter

Reply:
Manager feels too generic to me as a name - but again, I can't seem to name this functionality. Help!

Reply:
Manager as a whole is generic but it's within the context of connectionmanager: the consumer of the package will be typing connectionmanager.ConnectionManager repeatedly. But it could be connectionmanager.Manager or even connectiongroup.Manager